src/mol-plugin-state/helpers/drag-and-drop.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2023 mol* contributors, licensed under MIT, See LICENSE file for more info.
+ *
+ * @author David Sehnal <david.sehnal@gmail.com>
+ */
+
+
+import { OpenFiles } from '../actions/file';
+import { Asset } from '../../mol-util/assets';
+import { PluginCommands } from '../../mol-plugin/commands';
+import { PluginContext } from '../../mol-plugin/context';
+
+function defaultDragAndDropHandler(plugin: PluginContext, files: File[]) {
+    const sessions = files.filter(f => {
+        const fn = f.name.toLowerCase();
+        return fn.endsWith('.molx') || fn.endsWith('.molj');
+    });
+
+    if (sessions.length > 0) {
+        PluginCommands.State.Snapshots.OpenFile(plugin, { file: sessions[0] });
+    } else {
+        plugin.runTask(plugin.state.data.applyAction(OpenFiles, {
+            files: files.map(f => Asset.File(f)),
+            format: { name: 'auto', params: {} },
+            visuals: true
+        }));
+    }
+}
+
+export async function handleDragAndDrop(plugin: PluginContext, files: File[]) {
+    const handlers = Array.from(plugin.customDragAndDropHandlers.values());
+
+    for (const handler of handlers) {
```

Comment:
I would iterate in reverse order. So that newer handlers can simply override older handlers without having to remove them (use case: there is already a handler handling .pdb and .cif, I want to change behavior for .cif while keeping .pdb as is)

Reply:
Well, that's not so simple since the registration order isn't fully guaranteed. I will think about it a bit more.

Reply:
Refactored the code to a manager and applied your suggestion.

Reply:
Btw, I thought Map keeps the insertion order (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map).

Reply:
Apparently it does, but doesn't matter now.